src/engine/DisjunctionConstraint.h

```diff
@@ -144,6 +144,12 @@ class DisjunctionConstraint : public PiecewiseLinearConstraint
     */
     void getEntailedTightenings( List<Tightening> &tightenings ) const override;
 
+    /*
+      Transform the disjunction into a disjunction where each disjunct only
+      contains variable bounds.
+    */
+    void transformIfNeeded( InputQuery &inputQuery ) override;
```

Comment:
The function name should reflect what the function does, IfNeeded does not tell the reader much, functions are in general invoked when needed and no addition information is conveyed when it is needed. The latter can be discovered by searching where the function is used. 

Perhaps something along the lines of rewriteDisjunctionUsingAux / introduceAuxForDisjuncts?

Reply:
What you suggested is what I had, but Guy's suggested that we "have a general PLC method called from the Preprocessor, allowing PLCs to transform themselves.". I think the documentation should make it clear what this method does in the DisjunctionConstraint?

Reply:
changed to rewriteToUseAuxIfNeeded()